### PR TITLE
fix: support windows paths

### PR DESCRIPTION
**Pull request by andrei0x309**
Issue https://github.com/privatenumber/instant-mocha/issues/25

Quick patch to resolve module not found error on Win.

### CODE REVIEW

**Comment by privatenumber** [comment]
Thanks for the PR @andrei0x309 

Not sure if you can just omit the logic for Windows. I think you'll have to add the Windows version of resolution for it.

Also, we'll need tests that demonstrate this issue is fixed.

**Comment by andrei0x309** [comment]
@privatenumber Yeah this is meant to be a quick patch just to make it work on the Windows platform because for now the not found module error is 100% guaranteed on Windows, I saw that in the https://github.com/privatenumber/instant-mocha/issues/25 the user who reported the issue removed the line, I did the same and had no problem running the tests. So my idea was just to have a quick patch until you have something better.  I mean many users can still run the test under WSL but is much slower than windows if your files are outside of the WSL container. 

If you want I can try to write a test in the coming days for this quick patch.

**Comment by andrei0x309** [comment]
Yeah after some searches found that memfs works ok with `path.posix` functions on windows.

**Comment by amitzur** [comment]
@andrei0x309 Thanks for this!
I've just encountered the exact same issue, and meant to file a PR myself!
I was going to just do:
```
if (process.platform !== 'win32') {
  file = path.resolve(file)
}
```

The input is always going to be `/main.js`, no?

**Comment by andrei0x309** [comment]
@amitzur I think is better to use `path.posix.resolve` on windows than omitting it, if you omit it will likely work on windows with webpack 5, but it only passes 3 tests, if you use  `path.posix.resolve`  it will pass 18 test out of 20 tests. Also on WSL works with no modification but again it only passes 3 tests, only on native Linux it seems to pass all 20 tests included in the repo.

As far as detecting if it is win platform or not I guess you can do that in many ways.

File: src/lib/mocha.ts [hunk 1]
```diff
@@ -28,8 +28,11 @@ function createMochaInstance(
 		// Call with true tell mocha that we will be handling the file load outselves
 		this.lazyLoadFiles(true);
 
+		const isWin = path.delimiter === ';';
+
 		for (let file of files) {
-			file = path.resolve(file);
+			if (isWin) path.posix.resolve(file);
+			else file = path.resolve(file);
```
Comment (privatenumber) [inline src/lib/mocha.ts:35]: Remove if-condition and just use posix.resolve.

I believe this is doing "if windows, use posix.resolve, else (POSIX detected), use inferred posix.resolve"

**Comment by privatenumber** [comment]
For adding Windows tests, I think it's as simple as updating [this workflow](https://github.com/privatenumber/instant-mocha/blob/3880b1356e49e800fece6f3efd85f3bd847d793b/.github/workflows/test.yml#L12) to use a [`os` matrix](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-running-with-multiple-operating-systems) adding `windows-latest`.

**Comment by andrei0x309** [comment]
Yeah sorry, I didn't notice I didn't resign the file var. 

Tests on windows have same results 2 failed 18 passes, the test command from `package.json` on windows results with no tests found ran the tests on windows using ` node .\node_modules\jest\bin\jest.js`

**Comment by privatenumber** [comment]
Seems like the issue is with esm config loading. I currently don't have access to a Windows environment, but if you do, would you mind debugging the following?

```sh
$ cd tests/fixture
$ node ../../bin/instant-mocha.js --webpackConfig webpack.config.esm.mjs tests/passing-test.js
```

**Comment by andrei0x309** [comment]
image
Yeah, that's correct. 

I wrote a fix retested and the command: 
image

**Comment by privatenumber** [comment]
Thanks for all your help @andrei0x309

**Comment by amitzur** [comment]
Thanks! You're amazing

**Pull request merged**